src/openrct2-ui/windows/NewRide.cpp

```diff
@@ -818,18 +833,9 @@ static void WindowNewRideScrollpaint(rct_window* w, rct_drawpixelinfo* dpi, int3
                 dpi, { coords, coords + ScreenCoordsXY{ 115, 115 } }, w->colours[1], INSET_RECT_FLAG_FILL_MID_LIGHT | flags);
 
         // Draw ride image with feathered border
-        rideEntry = get_ride_entry(listItem->EntryIndex);
-        int32_t imageId = rideEntry->images_offset;
-
-        for (size_t i = 0; i < RCT2::ObjectLimits::MaxRideTypesPerRideEntry; i++)
-        {
-            if (rideEntry->ride_type[i] == listItem->Type)
-                break;
-
-            imageId++;
-        }
-
-        gfx_draw_sprite_raw_masked(dpi, coords + ScreenCoordsXY{ 2, 2 }, SPR_NEW_RIDE_MASK, imageId);
+        auto mask = ImageId(SPR_NEW_RIDE_MASK);
+        auto colour = GetRideImage(*listItem);
```

Comment:
`colour` seems a bit of an odd name for this.

```suggestion
        auto rideImage = GetRideImage(*listItem);
```